src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2190,18 +2190,32 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                         HashSet<DiagnosticInfo> useSiteDiagnostics = null;
                         bool reportedError = false;
                         foreach (var arm in switchExpression.SwitchArms)
-                        {
-                            var armConversion = this.Conversions.ClassifyImplicitConversionFromExpression(arm.Value, targetType, ref useSiteDiagnostics);
-                            if (!armConversion.IsImplicit || !armConversion.IsValid)
-                            {
-                                GenerateImplicitConversionError(diagnostics, arm.Value.Syntax, armConversion, arm.Value, targetType);
-                                reportedError = true;
-                            }
-                        }
+                            tryConversion(arm.Value, ref reportedError, ref useSiteDiagnostics);
```

Comment:
Consider wrapping in braces. All other one-liners in this function do, including a `while` loop above.